### ISSUE TITLE
diff chunks for certain unicode edits incorrect

### ISSUE THREAD

**Issue by kitsonk** [dtolnay/dissimilar#5]
I am getting unexpected result with `diff()` when using some unicode chars.  Specifically I think the problem is with the `🇺🇸` which is a combinatory sequence of combining 🇺 Regional Indicator Symbol Letter U and 🇸 Regional Indicator Symbol Letter S.

```rust
let a = " 🦕🇺🇸👍 ";
let b = " 🇺🇸👍 ";
let actual = diff(a, b);
assert_eq!(actual, vec![
  Chunk::Equal(" "),
  Chunk::Delete("🦕"),
  Chunk::Equal("🇺🇸👍 "),
]);
```

Which fails with:

```
thread panicked at 'assertion failed: `(left == right)`
  left: `[Equal(" "), Delete("🦕🇺"), Equal("🇸👍 ")]`,
 right: `[Equal(" "), Delete("🦕"), Equal("🇺🇸👍 ")]`'
```

I haven't dug into the code to see why it is breaking "incorrectly" across the unicode sequence, but wanted to raise the issue first.

**Comment by derekperkins** [comment]
Looks like this is also an issue with the original lib? https://github.com/google/diff-match-patch/issues/59

**Comment by dtolnay** [comment]
Fixed in 1.0.6.
